--- ax25.h.2.6	2010-12-03 13:32:33.416212744 -0500
+++ ax25.h	2010-12-03 13:35:30.456206697 -0500
@@ -34,6 +34,7 @@
  * For now, only UI frames without any Layer 3 protocol are handled.
  * This however is enough to send/receive APRS packets.
  *
+ * \version $Id$
  * \author Francesco Sacchi <batt@develer.com>
  *
  * $WIZ$ module_name = "ax25"
@@ -46,9 +47,10 @@
 #define NET_AX25_H
 
 #include "cfg/cfg_ax25.h"
+#include "ax25symbols.h"
 
 #include <cfg/compiler.h>
-#include <io/kfile.h>
+#include <kern/kfile.h>
 
 /**
  * Maximum size of a AX25 frame.
@@ -117,6 +119,7 @@ typedef struct AX25Msg
 	AX25Call dst;  ///< Destination address
 	#if CONFIG_AX25_RPT_LST
 	AX25Call rpt_lst[AX25_MAX_RPT]; ///< List of repeaters
+	bool rpt_used[AX25_MAX_RPT]; ///< Has this digi already passed this packet
 	uint8_t rpt_cnt; ///< Number of repeaters in this message
 	#endif
 	uint16_t ctrl; ///< AX25 control field
@@ -125,20 +128,6 @@ typedef struct AX25Msg
 	size_t len;    ///< Payload length
 } AX25Msg;
 
-#define AX25_CTRL_UI      0x03
-#define AX25_PID_NOLAYER3 0xF0
-
-/**
- * \name HDLC flags.
- * These should be moved in
- * a separated HDLC related file one day...
- * \{
- */
-#define HDLC_FLAG  0x7E
-#define HDLC_RESET 0x7F
-#define AX25_ESC   0x1B
-/* \} */
-
 
 /**
  * Declare an AX25 path.
